### PR TITLE
CI: New job for building the @automattic/wpcom-block-editor package

### PR DESCRIPTION
**Pull request by mmtr**
Part of #32044.

#### Changes proposed in this Pull Request

Add a new job to CircleCI in order to build the new `@automattic/wpcom-block-editor` package. The idea is to replicate the same build process for Masterbar Notifications, so we automatically build the package on each PR and store it as an artifact we can retrieve on WP.com to deploy a new version.

The job builds 2 versions of the package: minified and unminified. This is to ensure compliance of Jetpack with the [Guideline 8](https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/#8-plugins-may-not-send-executable-code-via-third-party-systems) for the plugin directory. JP sites will load the unminified script if `SCRIPT_DEBUG` is enabled (see p7jreA-2gu-p2 #comment-1887).

#### Testing instructions

* Check the details of the `build-wpcom-block-editor` this PR:
Screen Shot 2019-04-10 at 15 38 57
Screen Shot 2019-04-10 at 15 39 29

* You can also trigger a new one by clicking on the "Rebuild" button:
Screen Shot 2019-04-10 at 15 41 19

* Go to the Artifacts tab. You'll need to be logged in to CircleCI proper via GitHub for this tab to be visible.
* Make sure the both build files (minified and unminified) can be downloaded.

Screen Shot 2019-04-11 at 14 50 52

### CODE REVIEW

**Comment by kwight** [comment]
Is this fine like this, or does this need to be better aligned with the webpack approach by @ockham in #32083?

**Comment by mmtr** [comment]
> Is this fine like this, or does this need to be better aligned with the webpack approach by @ockham in #32083?

I'd say it's already a similar approach. The only difference is that for the notifications there is a `build-notifications` npm script, presumably because the notifications is not a package and we cannot use the `npx lerna run script --scope=<name>` syntax.

@ockham @sirreal do you have other thoughts?

File: .circleci/config.yml [hunk 1]
```diff
@@ -275,6 +276,19 @@ jobs:
...
+    <<: *defaults
+    parallelism: 1
+    steps:
+      - prepare
+      - run:
+          name: Build the block editor in WordPress.com integration utils package
+          command: |
+            npx lerna run clean --scope='@automattic/wpcom-block-editor'
+            npx lerna run bundle --scope='@automattic/wpcom-block-editor' -- -- --output-path=$CIRCLE_ARTIFACTS/wpcom-block-editor
+            NODE_ENV=production npx lerna run bundle --scope='@automattic/wpcom-block-editor' -- -- --output-path=$CIRCLE_ARTIFACTS/wpcom-block-editor --output-filename=[name].min.js
```
Comment (sirreal) [inline .circleci/config.yml]: I like this 😁

File: .circleci/config.yml [hunk 2]
```diff
@@ -275,6 +276,19 @@ jobs:
...
 
+  build-wpcom-block-editor:
+    <<: *defaults
+    parallelism: 1
+    steps:
+      - prepare
+      - run:
+          name: Build the block editor in WordPress.com integration utils package
+          command: |
+            npx lerna run clean --scope='@automattic/wpcom-block-editor'
```
Comment (sirreal) [inline .circleci/config.yml]: Clean is likely not necessary, but doesn't hurt 👍

File: .circleci/config.yml [hunk 3]
```diff
@@ -275,6 +276,19 @@ jobs:
...
+            npx lerna run bundle --scope='@automattic/wpcom-block-editor' -- -- --output-path=$CIRCLE_ARTIFACTS/wpcom-block-editor
```
Comment (sirreal) [inline .circleci/config.yml]: I was a bit surprised to see `bundle` and `build` here:

https://github.com/Automattic/wp-calypso/blob/33454d6e370e4df2935c4e21cbdb5d74a5340fbe/apps/wpcom-block-editor/package.json#L18-L23

I suspect that's more scripts than we need, but we can clean up as we iterate on `calypso-build`.

**Review by sirreal — APPROVED** [review]
The CI bits look good to me, I'll let @ockham answer https://github.com/Automattic/wp-calypso/pull/32163#issuecomment-482373134

**Comment by mmtr** [comment]
> The CI bits look good to me, I'll let @ockham answer [#32163 (comment)](https://github.com/Automattic/wp-calypso/pull/32163#issuecomment-482373134)

Thanks! I'll go ahead and merge this. We can address any feedback @ockham might have in a follow-up PR.

**Comment by ockham** [comment]
> > Is this fine like this, or does this need to be better aligned with the webpack approach by @ockham in #32083?
> 
> I'd say it's already a similar approach. The only difference is that for the notifications there is a `build-notifications` npm script, presumably because the notifications is not a package and we cannot use the `npx lerna run script --scope=<name>` syntax.
> 
> @ockham @sirreal do you have other thoughts?

I think it's fine the way you did it :+1: We might be able to drop the `build-notifications` script in a follow-up to #32083 if we decide to move notifications to the `apps/` dir.

**Pull request merged**